### ISSUE TITLE
Device is not ready yet when using unback to extract backup

### ISSUE THREAD

**Issue by rcmpayne** [libimobiledevice/ideviceinstaller#46]
1. Build the newest version of idevicebackup2you can use linux or winndows(mingw)
2. Run idevicebackup2 -d -i backup --full c:\backup
3. Backup completes without issue
4. try to unback idevicebackup2 -d -i unback c:\backup

Expected results:
the unback command is able to fully extract the backup

Actual results:
The backup will extract com.apple.Maps however, after that it starts to display "Device is not ready yet. Going to try again in 2 seconds..."

Logs:
http://pastebin.com/vz6KJbeG 

C:>idevicebackup2.exe -d -i unback C:\backup
Backup directory is "C:\backup"
This is an encrypted backup.
Enter backup password: **********
Started "com.apple.mobilebackup2" service on port 54946.
Negotiated Protocol Version 2.1
Reading Info.plist from backup.
Starting to unpack backup...
Backup password: Yes
Sending '340e42e15fc02fb99e7caf66565d3c881081390c/Manifest.plist' (11.1 KB)
Sending '340e42e15fc02fb99e7caf66565d3c881081390c/Manifest.mbdb' (145.4 KB)
Device is not ready yet. Going to try again in 2 seconds...
Device is not ready yet. Going to try again in 2 seconds...
Device is not ready yet. Going to try again in 2 seconds...

Environment:
Windows 7 with Mingw
iOS 9.3.1
backup encryption (tried with and without encryption)

**Comment by mcampetta** [comment]
Anyone assigned to or working on this bug? Replicated on most recent version as of 1/11/2018 with an iOS 10.1.2 device running on an iPhone 5. No backup encryption enabled. 

Sending '614604d744fced72b85bd44eb43615722e361efa/Manifest.plist' (13.1 KB)
Sending '614604d744fced72b85bd44eb43615722e361efa/Manifest.mbdb' (155.6 KB)
Device is not ready yet. Going to try again in 2 seconds...
Device is not ready yet. Going to try again in 2 seconds...
Device is not ready yet. Going to try again in 2 seconds...

**Comment by nikias** [comment]
Why do you file this under ideviceinstaller? This is a ticket for the main project. We have this issue filed already: https://github.com/libimobiledevice/libimobiledevice/issues/517
TL;DR: unback is broken since iOS 10.x - this is something Apple did and nothing that can be fixed from our side.

**Comment by paradoxicalNL** [comment]
Backup from IOS12, it unpacks like 2/3 of the files, after which it will fail.

Device is not ready yet. Going to try again in 2 seconds...

DMESG
[33976.576915] ipheth 1-2:4.2: ipheth_rcvbulk_callback: urb status: -84
[33976.577579] usb 1-2: USB disconnect, device number 17
[33976.605916] ipheth 1-2:4.2: Apple iPhone USB Ethernet now disconnected

on:

Copying 'a320bb6830e194f428ba0abe5ac146659d9d6f75/4d/4d6bb652961e1b6cefcf502caa08caea179e2b64' to '_unback_/a320bb6830e194f428ba0abe5ac146659d9d6f75/var/root/Library/Preferences/com.apple.xpc.activity2.plist'

In the meantime, most of my files are actually there!

**Comment by johnnyasantoss** [comment]
Mine only stops when it gets to `.../var/root/Library/Preferences/com.apple.xpc.activity2.plist`

Is there anything else I can do to unback my backup?

**Comment by mattschrage** [comment]
I have noticed the same thing. 

 `Copying '3a319fcdc444b81775a699113efc28fa00ee16f1/4d/4d6bb652961e1b6cefcf502caa08caea179e2b64' to '../var/root/Library/Preferences/com.apple.xpc.activity2.plist'
ERROR: Could not receive from mobilebackup2 (-256)
Unback Aborted.`

Any updates?

**Comment by ebangerter** [comment]
I have noticed the same thing. 

Copying '6b7ed6a498bed1619c242a579fe4cbf86f6afbd7/4d/4d6bb652961e1b6cefcf502caa08caea179e2b64' to '_unback_/6b7ed6a498bed1619c242a579fe4cbf86f6afbd7/var/root/Library/Preferences/com.apple.xpc.activity2.plist'
Device is not ready yet. Going to try again in 2 seconds...
Device is not ready yet. Going to try again in 2 seconds...
Device is not ready yet. Going to try again in 2 seconds...
